docs/reference/benchmarks.py

```diff
@@ -0,0 +1,178 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Code for generating the table in benchmarks.md."""
+
+import collections
+import multiprocessing
+import os
+import re
+import subprocess
+import sys
+import tarfile
+import zipfile
+
+from common import benchmark_utils
+from common import fuzzer_utils
+from common import utils
+from common import oss_fuzz
+
+BUILD_ARCHIVE_EXTENSION = '.tar.gz'
+LEN_BUILD_ARCHIVE_EXTENSION = len(BUILD_ARCHIVE_EXTENSION)
+COVERAGE_BUILD_PREFIX = 'coverage-build-'
+LEN_COVERAGE_BUILD_PREFIX = len(COVERAGE_BUILD_PREFIX)
+
+GUARDS_REGEX = re.compile(r'INFO:.*\((?P<num_guards>\d+) guards\).*')
+
+ONE_MB = 1024**2
+
+BENCHMARK_INFO_FIELDS = ['benchmark', 'target', 'dict', 'seeds', 'guards', 'MB']
+BenchmarkInfo = collections.namedtuple('BenchmarkInfo', BENCHMARK_INFO_FIELDS)
+
+
+def get_benchmark_infos(builds_dir):
+    """Get BenchmarkInfo for each benchmark that has a build in
+    builds_dir."""
+    build_paths = [
+        os.path.join(builds_dir, path)
+        for path in os.listdir(builds_dir)
+        if path.endswith(BUILD_ARCHIVE_EXTENSION)
+    ]
+    pool = multiprocessing.Pool()
+    return pool.map(get_benchmark_info, build_paths)
+
+
+def get_real_benchmark_name(benchmark):
+    """The method we use to infer benchmark names from coverage builds
+    doesn't quite work because the project name is used in OSS-Fuzz
+    builds instead. This function figures out the actual benchmark based on
+    the project name."""
+    benchmarks_dir = os.path.join(utils.ROOT_DIR, 'benchmarks')
+    real_benchmarks = [
```

Comment:
i think it might be better to iterate through directory and calculate the benchmark name based on if is_oss_fuzz or regular benchmark, this two loops - one here and later is more confusing.